titus_isolate/allocate/utils.py

```diff
@@ -0,0 +1,88 @@
+import os
+
+import boto3 as boto3
+
+from titus_isolate import log
+from titus_isolate.config.constants import MODEL_BUCKET_FORMAT_STR, MODEL_BUCKET_PREFIX, \
+    DEFAULT_MODEL_BUCKET_PREFIX, MODEL_BUCKET_LEAF, DEFAULT_MODEL_BUCKET_LEAF, MODEL_PREFIX_FORMAT_STR
+from titus_isolate.config.utils import get_required_property
+from titus_isolate.utils import get_config_manager
+
+
+def get_cpu_model_bucket_name():
+    format_str = get_required_property(MODEL_BUCKET_FORMAT_STR)
+    if format_str is None:
+        return None
+
+    region = os.environ['EC2_REGION']
```

Comment:
Any reason why you don't just "prerender" the bucket name as an env variable? And then leave it to some config file to figure this out?

Reply:
The current method is based on two priorities.
1. Don't expose actual bucket names and paths in the repository
2. Be able to redirect titus-isolate to download a model from an alternative location in an emergency

If bucket name is pre-rendered as part of the deb / systemd definition we leak some information.

Reply:
If you look at Titus-executor the way it's handled is by doing this:
https://github.com/Netflix/titus-executor/blob/master/root/apps/titus-executor/bin/run#L17-L22

And then can construct variables dynamically.

Reply:
I don't understand the benefit and this would appear to violate (1).